packages/block-library/src/navigation/style.scss

```diff
@@ -754,3 +788,20 @@ button.wp-block-navigation-item__content {
 html.has-modal-open {
 	overflow: hidden;
 }
+
+
+.wp-block-navigation__overlay {
+	display: none;
+	width: 100%; // fill the overlay
+	height: 100%; // fill the overlay
+}
+
+.is-menu-open.has-custom-overlay {
+	.wp-block-navigation__overlay {
+		display: initial;
+	}
+
+	.wp-block-navigation__default {
+		display: none;
+	}
+}
```

Comment:
We should comment this to explain what is happening here. Basically it's show/hiding the different content depending on whether it's being shown within the overlay.